api/api.js

```diff
@@ -1,93 +1,76 @@
+        };
+        client.db('task-service').collection("tasks").insertOne(task, (err, result) =>{
+            if(err){
+                console.log(err);
+            }
+    
+            res.send('task added successfully');
+        })
     })
-})
-
-//Updates categories
-app.patch('/db/task-service/categories/:_id',(req,res)=>{
-    Cat.findOneAndUpdate({_id: req.params.id},{
-        $set: req.body
-    }).then(()=>{
-        res.sendStatus(200);
+    
+    app.get('/db/task-service/tasks', (req,res)=>{
+        client.db('task-service').collection("tasks").find().toArray((err,results)=>{
+            res.send(results)
+        });
     });
-});
-
-//delete category
-app.delete('/db/task-service/categories/:_id',(req,res)=>{
-    Cat.findOneAndRemove({
-        _id: req.params.id
-    }).then((removedCatDoc)=>{
-        res.send(removedListDoc);
-    })
-})
-
-//Get all tasks for a category
-app.get('/db/task-service/tasks',(req,res)=>{
-    Task.find({
-        _catId: req.params.catId
-    }).then((tasks)=>{
-        res.send(tasks);
-    })
-});
-
-//get a single task
-app.post('/db/task-service/tasks/:_id',(req,res) => {
-    let newTask = new Task({
-        title: req.body.title,
-        _catId: req.params.catId
+    
+    app.get('/db/task-service/tasks:_id', (req,res,next)=>{
+        if(err){
+            throw err;
+        }
+        let _id=ObjectID(req.params._id);
+        client.db('task-service').collection("tasks").find(_id).toArray((err,results)=>{
+            if(err){
+                throw err;
+            }
+            res.send(results)
+        });
     });
-    newTask.save().then((newTaskDoc)=>{
-        res.send(newTaskDoc);
+    
+    app.put('/db/task-service/tasks:_id',(req,res,next)=>{
+        let id={
+            _id: ObjectID(req.params.id)
+        };
+        client.db('task-service').collection("tasks").updateOne({_id: id}, {$set:{'text':req.body.text, 'completed':req.body.completed}}, (err,result)=>{
+            if(err) {
+                throw err;
+              }
+            res.send('task updated sucessfully');
+        });
     });
-});
+    app.delete('/db/tasks-service/tasks:_id',(req,res,next)=>{
+        let _id=ObjectID(req.params.id);
+        client.db('task-service').collection("tasks").deleteOne(_id, (err,result)=>{
+            if(err){
+                throw err;
+            }
 
-//update exisiting task
-app.patch('/db/task-service/tasks/:_id',(req,res)=>{
-    Task.findOneAndUpdate({
-        _id:req.params.taskid,
-        _catId:req.params.catId,
-    },{
-        $set: req.body
-    }).then(()=>{
-        res.sendStatus(200);
+            res.send('task deleted');
+        })
     })
-})
-
-//delete task
-app.delete('/db/task-service/tasks/:_id',(req,res)=>{
-    Task.findOneAndRemove({
-    _id:req.params.taskId,
-    _listedId:req.params.listId
-    }).then((removedTaskDoc)=>{
-        res.send(removedTaskDoc);
+    app.listen(3000, ()=>{
```

Comment:
This is usually fine for personal projects but its good practice to have things like the Port number and DB connection string in a '.env' file at the root of the API folder.  This is to modularize data so we can easily deploy everything to services like AWS, Azure, or GCP.  Can you make this listen to the 'PORT' env?

See https://www.npmjs.com/package/dotenv for more information.

Reply:
I created a variable, I did try to setup a .env file but I was uncuessful in doing that. Paige went over a few examples in CLARK and CARD for how to do it